components/mdx/LessonLayout.tsx

```diff
@@ -31,6 +31,19 @@ const LessonLayout: React.FC<LayoutProps> = props => {
   const router = useRouter()
   const location = router.pathname.split('/')
   const current = location[location.length - 1]
+  const urlLessons = props.subLessons?.map(l =>
+    l.replace(',', '').toLowerCase().split(' ').join('_')
+  )
+  const index = urlLessons && urlLessons.indexOf(current)
+  let next
+  let previous
+  if (index && index > 0 && index < urlLessons!.length - 1) {
+    next = props.subLessons && props.subLessons[index + 1]
+    previous = props.subLessons && props.subLessons[index - 1]
+  }
+  if (index === 0) next = props.subLessons && props.subLessons[index + 1]
+  if (urlLessons && index === urlLessons.length - 1)
+    previous = props.subLessons && props.subLessons[index - 1]
```

Comment:
What's all this doing? It would be better to extract it into a function, there's too much logic inside the component code. By the function name other people can also have a better understanding of what it does.

Reply:
You're right I should have refactored this component long ago, it's hard to read. 

There are two things going on in this component. First layout component should decide which sub lesson to render as current, second it needs to create previous/next links to sub lessons.